src/read/reader.rs

```diff
@@ -261,10 +261,19 @@ pub trait Reader: Debug + Clone {
     /// Does not advance the reader.
     fn to_string_lossy(&self) -> Result<Cow<str>>;
 
+    /// Read a u8 slice.
+    fn read_slice(&mut self, buf: &mut [u8]) -> Result<()>;
```

Comment:
This always attempts to read `buf.len()` bytes and fails if it can't right? Does it panic if `buf.is_empty()`? Worth clarifying in the doc comment.